src/buffer.rs

```diff
@@ -34,6 +34,7 @@ pub struct PyBuffer<T>(Pin<Box<ffi::Py_buffer>>, PhantomData<T>);
 
 // PyBuffer is thread-safe: the shape of the buffer is immutable while a Py_buffer exists.
 // Accessing the buffer contents is protected using the GIL.
+#[allow(clippy::non_send_fields_in_send_ty)]
```

Comment:
See https://github.com/rust-lang/rust-clippy/issues/8045

We could also just make the fields `Send`, because right now the only types that implement `Element` are threadsafe already. However that would be a breaking change for anyone who has implemented `Element` on their own types.

Reply:
Good question. Should we mark this with a FIXME or TODO to decide later?

Reply:
Actually that's not going to silence the lint, because it's triggering on `ffi::Py_buffer`, and we can't make that Send. 🤔

Reply:
Ah ok. This is just one of the many cases where Python ffi makes clippy uneasy, we'll just have to live with it.